### ISSUE TITLE
Python 3.12 broken on 2.22.0

### ISSUE THREAD

**Issue by jay0lee** [googleapis/google-auth-library-python#1369]
This library breaks with Python 3.12.0rc1 on importing six.moves. I've not been able to find the root cause but six was already removed in #1346 and we just need a new release pushed out.

The [issue also surfaces](https://github.com/googleapis/google-auth-library-python-httplib2/issues/110) in google-auth-library-python-httplib2 so if anyone could look at that PR and get a release out there it'd be much appreciated.

Here's the relevant section of the error, I've confirmed there's no issue with 3.12 against current ToT so all we need for google-auth-python-library is a release.

```
import googleapiclient.discovery
  File "/opt/hostedtoolcache/Python/3.12.0-rc.1/x64/lib/python3.12/site-packages/googleapiclient/discovery.py", line 43, in <module>
    from google.auth.exceptions import MutualTLSChannelError
  File "/opt/hostedtoolcache/Python/3.12.0-rc.1/x64/lib/python3.12/site-packages/google/auth/__init__.py", line 20, in <module>
    from google.auth._default import (
  File "/opt/hostedtoolcache/Python/3.12.0-rc.1/x64/lib/python3.12/site-packages/google/auth/_default.py", line 30, in <module>
    import google.auth.transport._http_client
  File "/opt/hostedtoolcache/Python/3.12.0-rc.1/x64/lib/python3.12/site-packages/google/auth/transport/__init__.py", line 30, in <module>
    from six.moves import http_client
ModuleNotFoundError: No module named 'six.moves'
```

**Comment by clundin25** [comment]
Hi @jay0lee thanks for the heads up! When Python 3.12 is released we will update the documentation to indicate support, then do a release.

However a new release will likely be cut before then.

**Comment by clundin25** [comment]
We released v2.23.0 today, so this issue should be resolved.
